xmppserver/src/main/java/org/jivesoftware/openfire/interceptor/PacketRejectedException.java

```diff
@@ -94,11 +105,41 @@ public String getRejectionMessage() {
     /**
      * Sets the text to include in a message that will be sent to the sender of the packet
      * that got rejected or {@code null} if no message will be sent to the sender of the
-     * rejected packet. Bt default, no message will be sent.
+     * rejected packet. By default, no message will be sent.
      *
      * @param rejectionMessage the text to include in the notification message for the rejection.
      */
     public void setRejectionMessage(String rejectionMessage) {
         this.rejectionMessage = rejectionMessage;
     }
+
+    /**
+     * Gets the error that will be sent to the sender of the inbound stanza that got rejected.
+     *
+     * When the rejected stanza was a message, this error is added to a message that is sent (possibly also including
+     * the message body as specified by {@link #setRejectionMessage(String)}). When the rejected stanza was an IQ or
+     * Presence stanza, then the error is used in the error response. For these stanzas, this is a different stanza than
+     * the optional Message stanza that is sent when {@link #setRejectionMessage(String)}) is (also) used.
+     *
+     * @return An optional error to be included in an error response in reaction to the rejection.
+     */
+    public PacketError getRejectionError()
+    {
+        return rejectionError;
+    }
+
+    /**
+     * Adds an error that will be sent to the sender of the inbound stanza that got rejected.
+     *
+     * When the rejected stanza was a message, this error is added to a message that is sent (possibly also including
+     * the message body as specified by {@link #setRejectionMessage(String)}). When the rejected stanza was an IQ or
+     * Presence stanza, then the error is used in the error response. For these stanzas, this is a different stanza than
+     * the optional Message stanza that is sent when {@link #setRejectionMessage(String)}) is (also) used.
+     *
+     * @param rejectionError An optional error.
+     */
+    public void setRejectionError(PacketError rejectionError)
```

Comment:
Where is `setRejectionError` called?

Reply:
It is not used by Openfire itself. This is the new API that is the goal of this change, to be used by third parties / in plugins.